lewiskyron/Yank
# End-to-End Testing for Browser Extension Integration with Web Application

## Description

This pull request (PR) introduces the first end-to-end test connecting the web extension and the web app. With this update, users can now:
- Retrieve folders created in the database via the extension.
- Add flashcards to specific folders, storing them in both the `Folder` and `Flashcards tables` in the database.
- Access and view flashcards created through the extension on the web app.

### Key Points
Temporary User ID Handling: Currently, the `user ID (UUID)` is directly passed in the API call when creating flashcards due to the lack of authentication in the extension. Authentication will be added in future updates.

### Database Updates
Tables Added: Folder and Flashcards tables have been created in the Supabase PostgreSQL database to support these new functionalities.

## Type of Change


## Testing Instructions

To test this open the extension and highlight text. Select a folder then save the highlighted text. Open the Web app and you should be able to fetch the created cards on the web app. 

## Screenshots (if applicable)

No UI changes have been made.
